core/templates/dev/head/pages/creator-dashboard-page/creator-dashboard-page.controller.ts

```diff
@@ -177,41 +175,15 @@ angular.module('oppia').directive('creatorDashboardPage', [
             return value;
           };
 
-          var _fetchMessages = function(threadId) {
-            $http.get('/threadhandler/' + threadId).then(function(response) {
-              var allThreads = ctrl.mySuggestionsList.concat(
-                ctrl.suggestionsToReviewList);
-              for (var i = 0; i < allThreads.length; i++) {
-                if (allThreads[i].threadId === threadId) {
-                  allThreads[i].setMessages(response.data.messages);
-                  break;
-                }
-              }
-            });
-          };
-
           ctrl.clearActiveThread = function() {
             ctrl.activeThread = null;
           };
 
           ctrl.setActiveThread = function(threadId) {
-            _fetchMessages(threadId);
-            for (var i = 0; i < ctrl.mySuggestionsList.length; i++) {
-              if (ctrl.mySuggestionsList[i].threadId === threadId) {
-                ctrl.activeThread = ctrl.mySuggestionsList[i];
-                ctrl.canReviewActiveThread = false;
-                break;
-              }
-            }
-            if (!ctrl.activeThread) {
-              for (var i = 0; i < ctrl.suggestionsToReviewList.length; i++) {
-                if (ctrl.suggestionsToReviewList[i].threadId === threadId) {
-                  ctrl.activeThread = ctrl.suggestionsToReviewList[i];
-                  ctrl.canReviewActiveThread = true;
-                  break;
-                }
-              }
-            }
+            ctrl.activeThread = ctrl.threadsById[threadId];
```

Comment:
Do we need a check along the lines of threadsById.hasOwnProperty(threadId)?

Reply:
I don't think we don't but it can't hurt!